src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -170,13 +174,13 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
 
     deferredUpdates,
 
-    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | I | TI | null) {
+    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | PI | null) {
       const root : FiberRoot = (container.stateNode : any);
       const containerFiber = root.current;
       if (!containerFiber.child) {
         return null;
       }
-      return containerFiber.child.stateNode;
+      return getPublicInstance(containerFiber.child.stateNode);
```

Comment:
https://github.com/facebook/react/pull/8751/files#diff-7002876230f7effe9aa22e292f6a52c6R463 handles it in CommitWork. Added the switch for this method as well to do the same for `getRootPublicInstance`. Some tests have regressed that I’ll need to look into why.

Shouldn’t `render(<Foo />)` pass the `I | TI` back for backwards compatibility?